src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedAutoPropAccessorSymbol.cs

```diff
@@ -0,0 +1,205 @@
+            DiagnosticBag diagnostics)
+        {
+            _property = property;
+            _accessorKind = accessorKind;
+            Name = SourcePropertyAccessorSymbol.GetAccessorName(
+                paramName,
+                getNotSet: accessorKind == AccessorKind.Get,
+                // https://github.com/dotnet/roslyn/issues/44684
+                isWinMdOutput: false);
+
+            if (accessorKind != AccessorKind.Get)
+            {
+                var comp = property.DeclaringCompilation;
+                var type = TypeWithAnnotations.Create(comp.GetSpecialType(SpecialType.System_Void));
+
+                if (accessorKind == AccessorKind.Init)
+                {
+                    var initOnlyType = Binder.GetWellKnownType(
+                        comp,
+                        WellKnownType.System_Runtime_CompilerServices_IsExternalInit,
+                        diagnostics,
+                        property.Location);
+
+                    var modifiers = ImmutableArray.Create(CSharpCustomModifier.CreateRequired(initOnlyType));
+                    type = type.WithModifiers(modifiers);
+                }
+                _voidType = type;
+            }
+        }
+
+        public override TypeWithAnnotations ReturnTypeWithAnnotations
+        {
+            get
+            {
+                if (_voidType is object)
+                {
+                    Debug.Assert(MethodKind == MethodKind.PropertySet);
+                    return _voidType.Value;
+                }
+
+                return _property.TypeWithAnnotations;
+            }
+        }
+
+        public override MethodKind MethodKind => _accessorKind == AccessorKind.Get
+            ? MethodKind.PropertyGet
+            : MethodKind.PropertySet;
+
+        internal override bool IsInitOnly => _accessorKind == AccessorKind.Init;
+
+        public override int Arity => 0;
+
+        public override bool IsExtensionMethod => false;
+
+        public override bool HidesBaseMethodsByName => false;
+
+        public override bool IsVararg => false;
+
+        public override bool ReturnsVoid => _voidType is object;
+
+        public override bool IsAsync => false;
+
+        public override RefKind RefKind => RefKind.None;
+
+        public override FlowAnalysisAnnotations ReturnTypeFlowAnalysisAnnotations => FlowAnalysisAnnotations.None;
+
+        public override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotations => ImmutableArray<TypeWithAnnotations>.Empty;
+
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
+
+        public override ImmutableArray<ParameterSymbol> Parameters
+        {
+            get
+            {
+                if (_accessorKind == AccessorKind.Get)
+                {
+                    return ImmutableArray<ParameterSymbol>.Empty;
+                }
+                return ImmutableArray.Create(SynthesizedParameterSymbol.Create(
```

Comment:
>SynthesizedParameterSymbol.Create [](start = 45, length = 33)

Really? We are going to create a new parameter symbol every time someone asks for parameters? The array should be cached.